_migration-assistant/migration-phases/planning-your-migration/handling-type-mapping-deprecation.md

```diff
@@ -0,0 +1,291 @@
+
+- Route documents with type `user` to the `new_users` index
+- Route documents with type `post` to the `new_posts` index
+
+### Merge all types into one index
+
+To combine multiple types into a single index, use the following configuration:
+
+```json
+[
+  {
+    "TypeMappingSanitizationTransformerProvider": {
+      "staticMappings": {
+        "activity": {
+          "user": "activity",
+          "post": "activity"
+        }
+      },
+      "sourceProperties": {
+        "version": {
+          "major": 6,
+          "minor": 8
+        }
+      }
+    }
+  }
+]
+```
+{% include copy.html %}
+
+### Drop specific types
+
+To migrate only the `user` type within the `activity` index and drop all documents/requests with types not directly specified, use the following configuration:
+
+```json
+[
+  {
+    "TypeMappingSanitizationTransformerProvider": {
+      "staticMappings": {
+        "activity": {
+          "user": "users_only"
+        }
+      },
+      "sourceProperties": {
+        "version": {
+          "major": 6,
+          "minor": 8
+        }
+      }
+    }
+  }
+]
+```
+{% include copy.html %}
+
+This configuration only migrates documents of type `user` and ignore other types in the `activity` index.
+
+### Keep original structure
+
+To migrate only specific types and keep the original structure:
+
+```JSON
+[
+  {
+    "TypeMappingSanitizationTransformerProvider": {
+      "regexMappings": [
+        {
+          "sourceIndexPattern": "(.*)",
+          "sourceTypePattern": ".*",
+          "targetIndexPattern": "$1"
+        }
+      ],
+      "sourceProperties": {
+        "version": {
+          "major": 6,
+          "minor": 8
+        }
+      }
+    }
+  }
+]
+```
+{% include copy.html %}
+
+This is equivalent to the merge all types into one index rule but using a pattern-based routing strategy.
+
+### Combining multiple strategies
+
+You can combine both static and regex-based mappings to handle different indexes or patterns in a single migration. For example, you might have one index that must use `staticMappings` and another that uses `regexMappings` to route all types by pattern.
+
+For each document, request, or metadata item (processed individually for bulk requests), the following steps are performed:
+
+1. The index is checked to see if it matches an entry in the static mappings.
+   - If matched, the type is checked against the index component of the static mappings entry.
+     - If the type matches, the mapping is applied, and the resulting index is the value of the type key.
+     - If the type doesn't match, the request/document/metadata is dropped and not migrated.
+2. If the index is not matched in the static mappings, the index-type combination is checked against each item in the regex mappings list, in order from first to last. If a match is found, the mapping is applied, the resulting index becomes the value of the type key, and no further regex matching is performed.
```

Comment:
"index-type" => "index/type"? In the second sentence, does the resulting index literally become the value of the type key? If not, we need to use a verb other than "becomes".